invenio_preservation_sync/api.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Invenio-Preservation-Sync is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Preservation Sync API abstraction for the PreservationInfo entity."""
+
+from dateutil.parser import parse
+from invenio_db import db
+from sqlalchemy.sql import text
+
+from invenio_preservation_sync.errors import PreservationAlreadyReceivedError
+from invenio_preservation_sync.models import PreservationInfo, PreservationStatus
+
+
+class PreservationInfoAPI(object):
+    """
+    API abstraction of a PreservationInfo entity.
+
+    This class provides an abstraction layer for interacting with PreservationInfo objects.
+    It encapsulates the functionality for creating, retrieving, and managing them.
+    """
+
+    @classmethod
+    def create(
+        cls,
+        record_id,
+        revision_id,
+        status,
+        harvest_timestamp=None,
+        archive_timestamp=None,
+        uri=None,
+        path=None,
+        event_id=None,
+        description=None,
+    ):
+        """Create a Preservation Information."""
+        status = PreservationInfoAPI._convert_status(status)
```

Comment:
use `self` instead? Or was it on purpose?
```suggestion
        status = self._convert_status(self, status)
```

Reply:
I suggested also we don't need to be able to convert statuses, there should be only one style allowed by the API